src/main/java/shop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+        Set<CartItemResponseDto> cartItemResponseDtos = cartItemMapper
+                .toCartItemResponseDtos(shoppingCart.getCartItems());
+        ShoppingCartResponseDto shoppingCartResponseDto = shoppingCartMapper
+                .toShoppingCartResponseDto(shoppingCart);
+        shoppingCartResponseDto.setCartItems(cartItemResponseDtos);
+        return shoppingCartResponseDto;
+    }
+
+    @Transactional
+    @Override
+    public ShoppingCartResponseDto addBookToShoppingCart(
+            Long id, AddBookToCartRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartById(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can`t get shopping cart by id: " + id));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        CartItem existingCartItem = cartItems.stream()
+                .filter(cartItem -> cartItem.getBook().getId().equals(requestDto.getBookId()))
+                .findFirst()
+                .orElse(null);
+        if (existingCartItem != null) {
+            existingCartItem.setQuantity(requestDto.getQuantity() + existingCartItem.getQuantity());
+        } else {
+            CartItem cartItem = new CartItem();
+            cartItem.setShoppingCart(shoppingCart);
+            cartItem.setQuantity(requestDto.getQuantity());
+            cartItem.setBook(bookRepository.findById(requestDto.getBookId())
+                    .orElseThrow(() -> new EntityNotFoundException("Can`t get book by id: "
+                            + requestDto.getBookId())));
+            cartItem = cartItemRepository.save(cartItem);
+            cartItems.add(cartItem);
+        }
+        shoppingCart.setCartItems(cartItems);
+        shoppingCart = shoppingCartRepository.save(shoppingCart);
+        ShoppingCartResponseDto shoppingCartResponseDto = shoppingCartMapper
+                .toShoppingCartResponseDto(shoppingCart);
+        shoppingCartResponseDto.setCartItems(cartItemMapper.toCartItemResponseDtos(cartItems));
+        return shoppingCartResponseDto;
+    }
+
+    @Override
+    public CartItemResponseDto updateQuantityBook(Long cartItemId,
+                                                  UpdateQuantityBookRequestDto requestDto) {
+        CartItem cartItem = cartItemRepository.findById(cartItemId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can`t get cart by id: " + cartItemId));
+        CartItem updated = cartItem;
+        updated.setQuantity(cartItem.getQuantity() + requestDto.getQuantity());
+        cartItemMapper.updateCartItem(cartItem, updated);
+        return cartItemMapper.toCartItemResponseDto(cartItemRepository.save(updated));
+    }
+
+    @Transactional
+    @Override
+    public void removeBookFromShoppingCart(Long cartItemId, Long bookId) {
+        CartItem cartItem = cartItemRepository.findById(cartItemId)
```

Comment:
You can implement this method in this way: 

find shop cart
find cartItem `findByIdAndShoppingCartId`
create method `removeItemFromCart(cartItem);` in ShoppingCart entity
```
public void removeItemFromCart(CartItem cartItem) {
        cartItems.remove(cartItem);
    }
```
call it